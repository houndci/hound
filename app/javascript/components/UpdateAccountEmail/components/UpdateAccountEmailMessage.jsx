import React from 'react';

export default class UpdateAccountEmailMessage extends React.Component {
  render() {
    if (this.props.addressChanged === null) {
      return null;
    } else if (this.props.addressChanged === true) {
      return (
        <p className="inline-flash inline-flash--success" data-role="flash">
          Email address updated!
        </p>
      );
    } else {
      return (
        <p className="inline-flash inline-flash--error">
          There was a problem updating your email. Please try again.
        </p>
      );
    }
  }
}
