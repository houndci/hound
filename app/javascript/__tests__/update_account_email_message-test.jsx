import UpdateAccountEmailMessage from '../components/UpdateAccountEmail/components/UpdateAccountEmailMessage';

it('renders appropriately', () => {
  const wrapper = shallow(
    <UpdateAccountEmailMessage
      addressChanged={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders appropriately', () => {
  const wrapper = shallow(
    <UpdateAccountEmailMessage
      addressChanged={true}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
