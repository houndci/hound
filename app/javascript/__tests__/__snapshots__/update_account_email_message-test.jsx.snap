exports[`test renders appropriately 1`] = `
<p
  className="inline-flash inline-flash--error">
  There was a problem updating your email. Please try again.
</p>
`;

exports[`test renders appropriately 2`] = `
<p
  className="inline-flash inline-flash--success"
  data-role="flash">
  Email address updated!
</p>
`;
